GMG/Scenes/ChordProgress/SegmentScrollView.swift

```diff
@@ -0,0 +1,106 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import Foundation
+import SwiftUI
+
+struct SegmentsScrollView: View {
+    let totalDuration: TimeInterval
+    let segmentSlices: [[ChordSegmentSlice]]
+    let currentChordCell: ChordCell?
+    let selectedChordCell: ChordCell?
+    let segmentHandlers: SegmentHandlers
+    let waveformHandlers: WaveformHandlers
+    let audioLevels: [Float]
+    let elapsedTime: TimeInterval
+    let isPlaying: Bool
+
+    @State private var isUserScrolling = false
+    @State private var isWaveformDragging = false
+
+    private var resolvedWaveformHandlers: WaveformHandlers {
+        WaveformHandlers(
+            onTap: waveformHandlers.onTap,
+            onDragStart: { time in
+                isWaveformDragging = true
+                waveformHandlers.onDragStart(time)
+            },
+            onDragChange: waveformHandlers.onDragChange,
+            onDragEnd: { time in
+                waveformHandlers.onDragEnd(time)
+                isWaveformDragging = false
+            }
+        )
+    }
+
+    private var shouldAutoScroll: Bool {
+        isPlaying && isUserScrolling == false && isWaveformDragging == false
+    }
+
+    private var currentSegmentIndex: Int? {
+        guard segmentSlices.isEmpty == false else { return nil }
+        let rawIndex = Int(elapsedTime / Constants.segmentDuration)
+        let clampedIndex = min(
+            max(0, rawIndex),
+            segmentSlices.count - 1
+        )
+        return clampedIndex
+    }
+
+    var body: some View {
+        ScrollViewReader { proxy in
+            ScrollView {
+                LazyVStack(spacing: Spacing.md) {
+                    ForEach(Array(segmentSlices.enumerated()), id: \.offset) { index, slices in
+                        Segment(
+                            index: index,
+                            totalDuration: totalDuration,
+                            chordSlices: slices,
+                            segmentDuration: Constants.segmentDuration,
+                            currentChordCell: currentChordCell,
+                            selectedChordCell: selectedChordCell,
+                            audioLevels: audioLevels,
+                            elapsedTime: elapsedTime,
+                            segmentHandlers: segmentHandlers,
+                            waveformHandlers: resolvedWaveformHandlers
+                        )
+                        .id(index)
+                    }
+                }
+                .safeAreaPadding(Spacing.md)
+                .safeAreaPadding(.bottom, 128)
+            }
+            .onScrollPhaseChange { _, newPhase in
+                isUserScrolling = newPhase == .interacting
+            }
+            .onChange(of: isPlaying) { _, isPlaying in
```

Comment:
`.onChange`에 매개변수 없이 사용할 수도 있어요.

```suggestion
            .onChange(of: isPlaying) {
```